### PR TITLE
PHOENIX-5552 Hive against Phoenix gets 'Expecting "RPAREN", got "L"' …

### PR DESCRIPTION
**Pull request by brfrn169**
…in Tez mode

### CODE REVIEW

File: phoenix-hive/src/main/java/org/apache/phoenix/hive/constants/PhoenixStorageHandlerConstants.java [hunk 1]
```diff
@@ -91,6 +92,7 @@
     public static final String DATE_PATTERN = "'?\\d{4}-\\d{2}-\\d{2}'?";
     public static final String TIMESTAMP_PATTERN = "'?\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2}:\\d{2}\\" +
             ".?\\d{0,9}'?";
+    public static final String BIGINT_PATTERN = "[0-9]+L";
```
Comment (joshelser) [inline phoenix-hive/src/main/java/org/apache/phoenix/hive/constants/PhoenixStorageHandlerConstants.java]: Is the `L` only allowed to be capitalized? Or would `10l` (lowercase 'L') also be valid from the Hive side?

File: phoenix-hive/src/test/java/org/apache/phoenix/hive/query/PhoenixQueryBuilderTest.java [hunk 2]
```diff
@@ -170,4 +175,67 @@ public void testBuildQueryWithNotNull() throws IOException {
...
+        final String COLUMN_INT = "Column_Int";
+        final String COLUMN_BIGINT2 = "Column_Bigint2";
+        final String tableName = "TEST_TABLE";
+        final String expectedQueryPrefix = "select /*+ NO_CACHE  */ * from " + tableName
+          + " where ";
+
+        JobConf jobConf = new JobConf();
+        List<String> readColumnList = Collections.emptyList();
+
+        {
```
Comment (joshelser) [inline phoenix-hive/src/test/java/org/apache/phoenix/hive/query/PhoenixQueryBuilderTest.java]: Can you get rid of the curly-braces? I think it's fine to have these three clauses in this one test method, but, if you'd like to have them more logically separated, separate test methods would also be fine.

**Comment by ankitsinghal** [comment]
Thanks @brfrn169 for the fix, It seems from the code below, that we are using FILTER_TEXT_CONF_STR instead of serialized filters expression from FILTER_EXPR_CONF_STR for the Tez engine. I don't know the reason for doing it differently for MR and Tez, is this possible to use Filter expression rather than the string while building where clause for tez engine as well? or if you think it is a lot of work then we can go with your current fix as well.

**Comment by brfrn169** [comment]
Thank you for the review. @joshelser @ankitsinghal 

>  It seems from the code below, that we are using FILTER_TEXT_CONF_STR instead of serialized filters expression from FILTER_EXPR_CONF_STR for the Tez engine. I don't know the reason for doing it differently for MR and Tez, is this possible to use Filter expression rather than the string while building where clause for tez engine as well?

It seems like we can do this. And tested it locally, it looks worked at least to address this issue (and passed the existing unit tests). But wondering if this change has some side effects or regressions. I'm also not sure the reason for doing it differently for MR and Tez. If we can use Filter expression in Tez mode, I think it's the best. Do we have a way to verify if this change is invalid or not?

**Comment by ankitsinghal** [comment]
> It seems like we can do this. And tested it locally, it looks worked at least to address this issue (and passed the existing unit tests). But wondering if this change has some side effects or regressions. I'm also not sure the reason for doing it differently for MR and Tez. If we can use Filter expression in Tez mode, I think it's the best. Do we have a way to verify if this change is invalid or not?

if the unit tests are passing and you can verify some of the hive DDL/DMLs with Date/timestamp/between/null are working fine on a small cluster , then I think we should be good to go with it.

**Comment by brfrn169** [comment]
Thank you for your comment. @ankitsinghal 
I will do that.

**Comment by brfrn169** [comment]
In the latest patch, I made PhoenixInputFormat use Filter expression (FILTER_EXPR_CONF_STR) in Tez mode. I tested it locally and date/timestamp/between/null worked fine. Could you please review it? @ankitsinghal @joshelser

**Comment by ankitsinghal** [comment]
That's great @brfrn169 , it's a lot of unused code. Thanks for working on it. let me try this locally and will commit.

**Pull request closed without merging**